pkg/commands/sso/root.go

```diff
@@ -108,17 +110,36 @@ func (c *RootCommand) Exec(in io.Reader, out io.Writer) error {
 	// For creating/updating a profile we set `prompt` because we want to ensure
 	// that another session (from a different profile) doesn't cause unexpected
 	// errors for the user flow. This forces a re-auth.
-	if c.InvokedFromProfileCreate || ForceReAuth {
+	switch {
+	case c.InvokedFromProfileCreate || ForceReAuth:
 		c.Globals.AuthServer.SetParam("prompt", "login select_account")
-	}
-	if c.InvokedFromProfileUpdate || c.InvokedFromProfileSwitch {
+	case c.InvokedFromProfileUpdate || c.InvokedFromProfileSwitch:
 		c.Globals.AuthServer.SetParam("prompt", "login")
 		if c.ProfileSwitchEmail != "" {
 			c.Globals.AuthServer.SetParam("login_hint", c.ProfileSwitchEmail)
 		}
 		if c.ProfileSwitchCustomerID != "" {
 			c.Globals.AuthServer.SetParam("account_hint", c.ProfileSwitchCustomerID)
 		}
+	default:
+		// Handle `fastly sso` being invoked directly.
+		p := profile.Get(profileName, c.Globals.Config.Profiles)
+		if p == nil {
+			err := fmt.Errorf(profile.DoesNotExist, profileName)
+			c.Globals.ErrLog.Add(err)
+			return err
+		}
```

Comment:
This seems a bit harsh, couple of options to consider:

- Easiest: update the `profile.DoesNotExist` error message to also include a note on how they create one, currently it just says `the profile '%s' does not exist`, i.e. can we add `to create one, run: fastly profile create`
- Instead of erroring here, could we write a messaging saying that a profile doesn't exist and that we'll create one for them?
- Best: the above, but with a `Y/n` prompt